src/service/cache-cid-api.js

```diff
@@ -0,0 +1,209 @@
+};
+const API_KEYS = {
+  'googleanalytics': 'AIzaSyA65lEHUEizIsNtlbNo-l2K18dT680nsaM',
+};
+
+/**
+ * The Client ID service key.
+ * @const @private {string}
+ */
+const SERVICE_KEY_ = 'AIzaSyDKtqGxnoeIqVM33Uf7hRSa3GJxuzR7mLc';
+
+const TAG = 'CacheCidApi';
+const CACHE_API_URL = 'https://ampcid.google.com/v1/cache:getClientId?key=';
+
+const TIMEOUT = 30000;
+
+/**
+ * Exposes CID API for cache-served pages without a viewer.
+ */
+export class CacheCidApi {
+
+  constructor(ampdoc) {
+
+    /** @private {!./ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private {!./viewer-impl.Viewer} */
+    this.viewer_ = Services.viewerForDoc(this.ampdoc_);
+
+    /** @private {?Object<string, string>} */
+    this.apiKeyMap_ = null;
+
+    /** @private {?Promise<?string>} */
+    this.publisherCidPromise_ = null;
+
+    /** @private {!./timer-impl.Timer} */
+    this.timer_ = Services.timerFor(this.ampdoc_.win);
+  }
+
+  /**
+   * Returns true if the page is embedded in CCT and is served by a proxy.
+   * @returns {boolean}
+   */
+  isSupported() {
+    return this.isCctEmbedded() && this.viewer_.isProxyOrigin();
+  }
+
+  /**
+   * Returns scoped CID retrieved from the Viewer.
+   * @param {string} scope
+   * @return {!Promise<?string>}
+   */
+  getScopedCid(scope) {
+    // promise for whether we do stuff at all
+    if (!this.viewer_.isCctEmbedded()) {
+      return Promise.resolve(null);
+    }
+
+    const apiKey = this.isScopeOptedIn(scope);
+    if (!apiKey) {
+      return Promise.resolve(null);
+    }
+
+    if (!this.publisherCidPromise_) {
+      const url = CACHE_API_URL + SERVICE_KEY_;
+      this.publisherCidPromise_ = this.fetchCid_(url);
+    }
+
+    return this.publisherCidPromise_.then(publisherCid => {
+      return this.scopeCid_(publisherCid, scope);
+    });
+  }
+
+  /**
+   * Returns scoped CID retrieved from the Viewer.
+   * @param {string} url
+   * @return {!Promise<?string>}
+   */
+  fetchCid_(url) {
+    const payload = dict({
+      'publisherOrigin': getSourceOrigin(this.ampdoc_.win.location),
+    });
+
+    // Make the XHR request to the cache endpoint.
+    return this.timer_.timeoutPromise(
+        TIMEOUT,
+        Services.xhrFor(this.ampdoc_.win).fetchJson(url, {
+          method: 'POST',
+          ampCors: false,
+          credentials: 'include',
+          mode: 'cors',
+          body: payload,
+        }).then(res => {
+	  return res.json().then(response => {
+	    if (response['optOut']) {
+	      return null;
+	    }
+	    const cid = response['publisherClientId'];
+            if (!cid && response['alternateUrl']) {
+              // If an alternate url is provided, try again with the alternate url
+              // The client is still responsible for appending API keys to the URL.
+              const alt = `${response['alternateUrl']}?key=${SERVICE_KEY_}`;
+              return this.fetchCid_(dev().assertString(alt))
+                  .then(altRes => {
```

Comment:
Rather than dupe this logic, let's add an optional param to `fetchCid_` that checks for `alternateUrl` and defaults to `true`. Then pass `false` for this invocation.